### ISSUE TITLE
missing security options in na_ontap_vserver_cifs module, na_ontap_vserver_cifs_security deprecated

### ISSUE THREAD

**Issue by ansibleguy76** [ansible-collections/netapp.ontap#187]
### Summary

Setting is_aes_encryption_enabled, lm_compatibility_level, ... is not possible,

### Component Name

na_ontap_vserver_cifs

### Ansible Version

```console
ansible [core 2.14.5]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.11/site-packages/ansible
  ansible collection location = /usr/share/ansible/collections/ansible_collections:/etc/ansible/collections/ansible_collections
  executable location = /usr/bin/ansible
  python version = 3.11.5 (main, Aug 26 2023, 11:59:23) [GCC 12.2.1 20220924] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

### ONTAP Collection Version

```console
# /usr/share/ansible/collections/ansible_collections
Collection         Version
------------------ -------
amazon.aws         6.3.0
community.general  7.3.0
netapp.elementsw   21.7.0
netapp.ontap       22.7.0
netapp.storagegrid 21.11.1
netapp.um_info     21.8.0

# /etc/ansible/collections/ansible_collections
Collection   Version
------------ -------
netapp.ontap 22.7.0

# /usr/lib/python3.11/site-packages/ansible_collections
Collection                    Version
----------------------------- -------
amazon.aws                    5.4.0
ansible.netcommon             4.1.0
ansible.posix                 1.5.2
ansible.utils                 2.9.0
ansible.windows               1.13.0
arista.eos                    6.0.1
awx.awx                       21.14.0
azure.azcollection            1.15.0
check_point.mgmt              4.0.0
chocolatey.chocolatey         1.4.0
cisco.aci                     2.6.0
cisco.asa                     4.0.0
cisco.dnac                    6.7.1
cisco.intersight              1.0.27
cisco.ios                     4.5.0
cisco.iosxr                   4.1.0
cisco.ise                     2.5.12
cisco.meraki                  2.15.1
cisco.mso                     2.4.0
cisco.nso                     1.0.3
cisco.nxos                    4.3.0
cisco.ucs                     1.8.0
cloud.common                  2.1.3
cloudscale_ch.cloud           2.2.4
community.aws                 5.4.0
community.azure               2.0.0
community.ciscosmb            1.0.5
community.crypto              2.12.0
community.digitalocean        1.23.0
community.dns                 2.5.3
community.docker              3.4.3
community.fortios             1.0.0
community.general             6.6.0
community.google              1.0.0
community.grafana             1.5.4
community.hashi_vault         4.2.0
community.hrobot              1.8.0
community.libvirt             1.2.0
community.mongodb             1.5.2
community.mysql               3.6.0
community.network             5.0.0
community.okd                 2.3.0
community.postgresql          2.3.2
community.proxysql            1.5.1
community.rabbitmq            1.2.3
community.routeros            2.8.0
community.sap                 1.0.0
community.sap_libs            1.4.1
community.skydive             1.0.0
community.sops                1.6.1
community.vmware              3.5.0
community.windows             1.12.0
community.zabbix              1.9.3
containers.podman             1.10.1
cyberark.conjur               1.2.0
cyberark.pas                  1.0.17
dellemc.enterprise_sonic      2.0.0
dellemc.openmanage            6.3.0
dellemc.os10                  1.1.1
dellemc.os6                   1.0.7
dellemc.os9                   1.0.4
dellemc.powerflex             1.6.0
dellemc.unity                 1.6.0
f5networks.f5_modules         1.23.0
fortinet.fortimanager         2.1.7
fortinet.fortios              2.2.3
frr.frr                       2.0.2
gluster.gluster               1.0.2
google.cloud                  1.1.3
grafana.grafana               1.1.1
hetzner.hcloud                1.11.0
hpe.nimble                    1.1.4
ibm.qradar                    2.1.0
ibm.spectrum_virtualize       1.11.0
infinidat.infinibox           1.3.12
infoblox.nios_modules         1.4.1
inspur.ispim                  1.3.0
inspur.sm                     2.3.0
junipernetworks.junos         4.1.0
kubernetes.core               2.4.0
lowlydba.sqlserver            1.3.1
mellanox.onyx                 1.0.0
microsoft.ad                  1.0.0
netapp.aws                    21.7.0
netapp.azure                  21.10.0
netapp.cloudmanager           21.22.0
netapp.elementsw              21.7.0
netapp.ontap                  22.5.0
netapp.storagegrid            21.11.1
netapp.um_info                21.8.0
netapp_eseries.santricity     1.4.0
netbox.netbox                 3.12.0
ngine_io.cloudstack           2.3.0
ngine_io.exoscale             1.0.0
ngine_io.vultr                1.1.3
openstack.cloud               1.10.0
openvswitch.openvswitch       2.1.0
ovirt.ovirt                   2.4.1
purestorage.flasharray        1.17.2
purestorage.flashblade        1.11.0
purestorage.fusion            1.4.2
sensu.sensu_go                1.13.2
splunk.es                     2.1.0
t_systems_mms.icinga_director 1.32.2
theforeman.foreman            3.10.0
vmware.vmware_rest            2.3.1
vultr.cloud                   1.7.0
vyos.vyos                     4.0.2
wti.remote                    1.0.4
```

### ONTAP Version

```console
NetApp Release 9.11.1: Tue Jul 12 04:38:28 UTC 2022
```

### Playbook

```console
- name: "Set default security settings"      
  netapp.ontap.na_ontap_vserver_cifs_security:
    vserver        : "{{ svm_name }}"
    is_aes_encryption_enabled: false
    lm_compatibility_level: ntlmv2_krb
    <<             : *auth
  # ignore_errors: true
```

### Steps to Reproduce

```yaml (paste below)

- name: "Set default security settings"      
  netapp.ontap.na_ontap_vserver_cifs_security:
    vserver        : "{{ svm_name }}"
    is_aes_encryption_enabled: false
    lm_compatibility_level: ntlmv2_krb
    <<             : *auth
  # ignore_errors: true
```

### Expected Results

No issues, working rest

### Actual Results

```console
[WARNING]: Module did not set no_log for is_password_complexity_required 2023-11-07 18:55:55
[WARNING]: This module only supports ZAPI and is deprecated.   It will no
longer work with newer versions of ONTAP.  The final ONTAP version to support
ZAPI is ONTAP 9.12.1.
[WARNING]: netapp.ontap.na_ontap_cifs_server should be used instead.
[WARNING]: With version 22.0.0 ONTAPI (ZAPI) has been deprecated. The final
ONTAP version to support ZAPI is ONTAP 9.13.1.  ZAPI calls in these modules
will continue to work for ONTAP versions that supports ZAPI.  You can update
your playbook to use REST by adding use_rest: always to your playbook.  More
information can be found at: https://github.com/ansible-
collections/netapp.ontap
```
==> only supported with zapi, suggested to use na_ontap_cifs_server
==> na_ontap_cifs_server does not have half of the security settings

**Comment by carchi8py** [comment]
The ONTAPI variable `is-aes-encryption-enabled` was renamed to `security.kdc_encryption` in the rest API and can be set with `kdc_encryption`
https://docs.ansible.com/ansible/devel/collections/netapp/ontap/na_ontap_cifs_server_module.html#parameter-kdc_encryption

We don't have REST equivalent for `lm-compatibility-level` I'll add a story to add that. 

You can use this page as a reference to see if variables have been renamed in the REST API
https://docs.netapp.com/us-en/ontap-restmap-9131/cifs.html#cifs-security-modify

**Comment by carchi8py** [comment]
We have DEVOPS-6529 open for adding lm-compatibility-level  to the na_ontap_cifs_server module

**Comment by carchi8py** [comment]
This was fixed in 22.9.0
